src/test/java/org/kohsuke/github/GHPullRequestTest.java

```diff
@@ -883,6 +883,41 @@ public void checkPullRequestReviewer() throws IOException {
         assertThat(reviewer, notNullValue());
     }
 
+    /**
+     * Test refreshing a PR coming from the search results.
+     *
+     * @throws Exception
+     *             the exception
+     */
+    @Test
+    public void refreshFromSearchResults() throws Exception {
+        String prName = "refreshFromSearchResults";
+        GHRepository repository = getRepository();
+
+        repository.createPullRequest(prName, "test/stable", "main", "## test");
+
+        // we need to wait a bit for the pull request to be indexed by GitHub
+        Thread.sleep(2000);
+
+        GHPullRequest pullRequestFromSearchResults = repository.searchPullRequests()
+                .isOpen()
+                .titleLike(prName)
+                .list()
+                .toList()
+                .get(0);
+
+        pullRequestFromSearchResults.getMergeableState();
+
+        // wait a bit for the mergeable state to get populated
+        Thread.sleep(5000);
```

Comment:
Please implement `await()` in some form instead of hard sleeps.  Waiting is only needed when recording, not during CI tests. 

https://github.com/hub4j/github-api/blob/main/src/test/java/org/kohsuke/github/GHWorkflowRunTest.java#L574-L584

Reply:
This is broader problem.  Filed #1815.

Reply:
Yeah I agree on that. I actually asked myself that and saw that there was some `Thread.sleep`  calls above so I supposed we decided to live with them. I'll try to find some cycles to clean this up.